src/UI/Component/MainControls/Factory.php

```diff
@@ -0,0 +1,271 @@
+	 *     are "Search", "Help", "Notifications", "Awareness" and "User"
+	 *
+	 *     Especially in mobile context, the total width of all entries may exceed
+	 *     the availble width of the screen. In this case, all entries are
+	 *     summarized under a "..."-Button. The logo remains.
+	 *
+	 *     Elements are rendered as Bulky Buttons. Prompts in the Metabar may be
+	 *     marked with counters for new/existing notifications.
+	 *
+	 *   effect: >
+	 *     Entries in the Metabar may open a Slate when clicked. They will be set
+	 *     to "engaged" accordingly, and bear the aria-pressed attribute. There
+	 *     will be only one engaged Button/Slate at a time. Also, Buttons in the
+	 *     Metabar may trigger navigation or activate tools in the Mainbar, like
+	 *     the Help. In this case, the buttons are not stateful.
+	 *
+	 *   rivals:
+	 *     Mainbar: >
+	 *       The Mainbar offers navigational strategies, while the Metabar foremost
+	 *       provides notifications to the user and offers controls that are deemed
+	 *       important.
+	 *       The (general) direction of communication for the Metabar is "system to
+	 *       user", while the direction is "user to system" for elements of the Mainbar.
+	 *
+	 * context:
+	 *    - The Metabar is used in the Standard Page.
+	 *
+	 * rules:
+	 *   usage:
+	 *     1: The Metabar is unique for the page - there MUST be at most one.
+	 *     2: Elements in the Metabar MUST NOT vary according to context.
+	 *     3: New elements in the Metabar MUST be approved by JF.
+	 *
+	 *   style:
+	 *     1: The bar MUST have a fixed height.
+	 * ----
+	 *
+	 * @return  \ILIAS\UI\Component\MainControls\Metabar
+	 */
+	public function metabar(Image $logo): Metabar;
+
+	/**
+	 * ---
+	 * description:
+	 *   purpose: >
+	 *     The Mainbar is a unique page section that bundles access to content-
+	 *     based navigational strategies (like the repository tree) as well as
+	 *     navigation to services unrelated to the actual content, like the
+	 *     administrative settings.
+	 *
+	 *     Since the controls necessary for theses purposes might be quite complex,
+	 *     they are summed up in an easy to grasp Icon or Glyph in conjunction with
+	 *     a short text. Theses reductions form the entries for the mainbar, which
+	 *     thus is the primary list of navigational options for the user and the
+	 *     usual starting point for the user to explore the system.
+	 *
+	 *     There are entries in the bar that are never modified by changing context,
+	 *     but may vary according to e.g. the current user's permissions or settings
+	 *     of the installation. There also is the tools-section of entries in the bar
+	 *     that is used to show tools that are opened on request of the user, e.g. the
+	 *     help, or depending on requirements of the content, e.g. a local navigation.
+	 *
+	 *   composition: >
+	 *     The Mainbar holds Slates and Bulky Buttons.
```

Comment:
In the metabar the entries (bulky buttons) open slates. Why not the same in the mainbar?

Reply:
This is very similar to the metabar. In fact we have anticipated the requirement that some Bulky Button in the Mainbar just opens some page (instead of a slate) here.